sklearn/ensemble/_stacking.py

```diff
@@ -592,9 +608,21 @@ def fit(self, X, y, sample_weight=None):
             Returns a fitted instance of estimator.
         """
         check_classification_targets(y)
-        self._le = LabelEncoder().fit(y)
-        self.classes_ = self._le.classes_
-        return super().fit(X, self._le.transform(y), sample_weight)
+        self._type_of_target = type_of_target(y)
+        self.classes_ = []
+        if self._type_of_target != "multilabel-indicator":
+            self._label_encoder = LabelEncoder().fit(y)
+            self.classes_ = self._label_encoder.classes_
+            y = self._label_encoder.transform(y)
```

Comment:
```python
if type_of_target(y) == "multilabel-indicator":
    self._label_encoder = [LabelEncoder().fit(yk) for yk in y.T]
    self.classes_ = [le.classes_ for le in self._label_encoder]
    y_encoded = np.array(
        self._label_encoder[target_idx].transform(target)
        for target in y.T
    )
else:
    self._label_encoder = LabelEncoder().fit(y)
    self.classes_ = self._label_encoder.classes_
    y_encoded = self._label_encoder.transform(y)
return super().fit(X, y_encoded, sample_weight)
```

Reply:
We should not modify `y` in-place as it is the case for the multilabel case now. It will surprise our user that can loose is original `y`.

We shoul add a test to check this behavior does not happen by checking that `y` is not alterated after the `fit` of the stacking classifier.

Reply:
We should as well update the docstring of `classes_` that can be a list of numpy arrays is usually done in the multilabel case.